### PR TITLE
Make output_keys param MultiOption

### LINKED ISSUE #6676: [CT-1869] Turn `--output-keys` into MultiOption
### Current state

Users can supply `--output-keys` to customize the set of information returned by `dbt list`:
```bash
$ dbt ls --output json --output-keys name,resource_type,description
{"name": "my_incremental_model", "resource_type": "model", "description": "cool docs bro"}
{"name": "my_seed", "resource_type": "seed", "description": ""}
```

The type of `--output-keys` is a string. I've always thought the string needed to be a comma-delimited list, but really it could be ... anything? ... so long as the string contains the relevant key name:

https://github.com/dbt-labs/dbt-core/blob/c85be323f537ff31991e618319bbf9763e16b65c/core/dbt/cli/params.py#L157-L159

```
$ dbt ls --output json --output-keys nameNotACommaresource_type==blabladescription
{"name": "my_incremental_model", "resource_type": "model", "description": "cool docs bro"}
{"name": "my_seed", "resource_type": "seed", "description": ""}
```

### Proposal

We just (https://github.com/dbt-labs/dbt-core/pull/6641) added a [`MultiOption`](https://github.com/dbt-labs/dbt-core/blob/feature/click-cli/core/dbt/cli/options.py#L6) option to support params like `--select`, that take multiple space-limited arguments:
```
$ dbt run --select thing_a thing_b --another-flag ...
```

This would be more consistent, and allow us to treat the type of `output_keys` as a list (right?) for programmatic invocations:
```
$ dbt ls --output json --output-keys name resource_type description
```

Looks like that's even how we've been documenting it (incorrectly)! https://docs.getdbt.com/reference/commands/list

### Add a test

We also need an automated test for `dbt list --output json --output-keys`!

### PR DESCRIPTION
**Pull request by stu-k**
resolves #6676

### Description

Changes `output_keys` to a list of strings rather than a comma delimited string.

### Checklist

- [X] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me
- [X] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [X] I have run this code in development and it appears to resolve the stated issue
- [X] This PR includes tests, or tests are not required/relevant for this PR
- [X] I have [opened an issue to add/update docs](https://github.com/dbt-labs/docs.getdbt.com/issues/new/choose), or docs changes are not required/relevant for this PR
- [X] I have run `changie new` to [create a changelog entry](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: core/dbt/cli/params.py [hunk 1]
```diff
@@ -172,7 +172,12 @@
 )
 
 output_keys = click.option(
-    "--output-keys", envvar=None, help="TODO: No current help text", type=click.STRING
+    "--output-keys",
+    envvar=None,
+    help="TODO: No current help text",
```
Comment (stu-k) [inline core/dbt/cli/params.py:177]: @dbeatty10 Could I ask for your opinion on some help text here as well please 😅

Comment (dbeatty10) [reply core/dbt/cli/params.py:177]: See below on some ideas we can iterate on inspired by the docs on [listing JSON output with custom keys](https://docs.getdbt.com/reference/commands/list#:~:text=Listing%20JSON%20output%20with%20custom%20key):

Idea 1:
```
Space-delimited listing of which node properties are included in the output when `--output json`.
```

Idea 2:
```
Custom keys for JSON output when `--output json`. Space-delimited listing of node properties to include.
```

Idea 3:
```
Space-delimited listing of node properties to include as custom keys for JSON output.
```

If we include examples in the help text, then maybe something like this:
```
Space-delimited listing of node properties to include as custom keys for JSON output.
Ex. `--output json --output-keys "name resource_type description"`
```

Comment (stu-k) [reply core/dbt/cli/params.py:177]: I dig the last option with the example!

Comment (dbeatty10) [reply core/dbt/cli/params.py]: 🙌

**Review by MichelleArk — APPROVED** [review]

**Pull request merged**